### PR TITLE
Water refactor 2: refactor `soil/evapotranspiration.py`

### PR DESCRIPTION
**Pull request by Isgotkowitz**
# Summary
This PR overhauls the `soil/evapotranspiration.py` file that was originally refactored in late January/early February. Originally, the `Evapotranspiration` module attempted to handle all aspects of evaporation from the field. However this was an inappropriate design because evapotranspiration is a process that takes place over the field, not just the soil. This PR restricts the scope of the redesigned module `Evaporation` to only handle evaporation from the soil profile, as well as handles some side effects of treating evapotranspiration as a field-level process in the soil module.

# Changes
- `evapotranspiration.py` and `test_evapotranspiration.py` have been renamed to `evaporation.py` and `test_evaporation.py` respectively.
- ***Important*** all methods (and associated tests) besides `evaporate()` in `evaporation.py` do not need to be reviewed. They are unchanged, but GitHub shows them all as new because they are in a file that was renamed.
  - `evaporate()` handles evaporating from the soil profile, stores the amount evaporated on the current day, and adds to the total amount of evaporation that has occurred.
- `infiltration.py`
  - Because evapotranspiration is handled at the `Field` level now, the amount of potential evapotranspiration for the current day is passed to this module as a parameter.
  - The test for `infiltrate()` now reflects this as well.
- `soil_data.py`
  - All attributes that track evapotranspiration have been removed, and replaced with a pair of attributes that track how much water is evaporated from the soil profile on a given day and how much has been evaporated over the course of a year.
  - These changes are reflected in `test_annual_reset()` and `test_config_factory_defaults()`.
- `soil.py`
  - `Evapotranspiration` has been changed to `Evaporation`
  - The parameters for `daily_soil_water_routine()` have been updated so that unnecessary parameters for `Evaporation` have been removed, and `potential_evapotranspiration` has been added so that `Infiltration`'s main method can be run.

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/soil/evaporation.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        self.data.water_evaporated = total_evaporation_from_soil
+        self.data.annual_soil_evaporation_total += total_evaporation_from_soil
+        return
+
+    # TODO - this method should be moved to field.py and used there when sublimation is implemented #317
+    @staticmethod
+    def _determine_maximum_soil_evaporation(soil_evaporation_adj: float, snow_water_content: float) -> float:
+        """Calculates the maximum amount of evaporation from soil in a given day
+
+        Args:
```
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/soil/evaporation.py]: Minor point: Here and in some of the the following functions will need to be updated to adhere to the Sphinx documentation format, as done in the __init__ block and the `evaporate` method. e.g. "Parameters" instead og "Args", and typing in the docstring.

File: SC_redesign/Crop_and_Soil/soil/infiltration.py [hunk 2]
```diff
@@ -23,12 +23,14 @@ def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] =
         """
         self.data = soil_data or SoilData(field_size=field_size)
 
-    def infiltrate(self, rainfall: float, weighting_coefficient: float) -> None:
+    def infiltrate(self, rainfall: float, weighting_coefficient: float, potential_evapotranspiration: float) -> None:
         """main routine for determining runoff and infiltration of soil for a given day
 
         rainfall: rainfall depth of current day (mm)
```
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/soil/infiltration.py]: needs "Parameters" heading for docstring

**Review by JoeWaddell — APPROVED** [review]
Just left a few comments about formatting of the docstrings, but it otherwise looks good to me. The initial comment explaining this PR was especially helpful. Nice work!

File: SC_redesign/Crop_and_Soil/soil/evaporation.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            amount_water_removed = self._determine_amount_water_removed(
+                evaporative_demand_reduced, layer.water_content, layer.wilting_point_content)
+
+            maximum_amount_evaporated = amount_available_for_evaporation <= amount_water_removed
+            if maximum_amount_evaporated:
+                amount_water_removed = amount_available_for_evaporation
+            layer.water_content -= amount_water_removed
+            amount_available_for_evaporation -= amount_water_removed
+            if maximum_amount_evaporated:
+                break
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/evaporation.py]:
```suggestion
      if maximum_amount_evaporated:
            amount_water_removed = amount_available_for_evaporation
            layer.water_content -= amount_water_removed
            amount_available_for_evaporation -= amount_water_removed
            break
      else:
            layer.water_content -= amount_water_removed
            amount_available_for_evaporation -= amount_water_removed
```
Although there are more lines, I think that reducing condition statements improves readability. I don't think it's a big deal so up to you tho.

File: SC_redesign/Crop_and_Soil/soil/evaporation.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+        total_evaporation_from_soil = maximum_soil_water_evaporation - amount_available_for_evaporation
+        self.data.water_evaporated = total_evaporation_from_soil
+        self.data.annual_soil_evaporation_total += total_evaporation_from_soil
+        return
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/evaporation.py]: I'm not sure what's the need for this return.

**Review by matthew7838 — APPROVED** [review]
LGTM has a few suggestions. Good notes that make it a lot easier for reviewers to catch up!

File: SC_redesign/Crop_and_Soil/soil/evaporation.py (hunk 4 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/soil/evaporation.py]: There isn't any. I guess I just felt the desire for an extra return.

File: SC_redesign/Crop_and_Soil/soil/evaporation.py (hunk 3 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/soil/evaporation.py]: This is a good suggestion. I did not go with the duplicate lines but I found a way to accomplish the same thing with a `min` statement and a single `if` statement.

**Pull request merged**